src/lib.rs

```diff
@@ -38,18 +38,19 @@ impl Lehmer {
         Self { code }
     }
 
-    pub fn to_permutation(mut self) -> Vec<u64> {
-        let n = self.code.len() as u64;
+    pub fn to_permutation(&self) -> Vec<u64> {
+        let mut code = self.code.to_owned();
```

Comment:
Note this means calling `to_permutation` no longer updates a `Lehmer` instance's `code` in-place: I wasn't sure if this behaviour was intentional or not. If so, we can add tests to ensure that behaviour is preserved (or avoided).

Reply:
Hmm, this behaviour was actually intentional to make it a tiny bit faster:
https://github.com/tuzz/lehmer/commit/22166df5ebcc1eac10b5b27244a4eafd303ffd4f

My thinking was the person using this library could `.clone()` the vector first if they wanted to keep it, otherwise it can be happily taken over by this function to gain a little on speed.

The `cargo bench` shows a slight speed drop (~25%) for `#to_permutation`. Maybe instead of changing the signature, I should change the names of these methods? [This page](https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#wrong_self_convention) doesn't seem to have a convention for `mut self` though. 🤔

Reply:
Ah, then maybe this is just a naming convention thing: I'll put this change out of the PR.